core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -97,16 +100,66 @@
       new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
           RelFactories.LOGICAL_BUILDER);
 
+  private final Set<SqlKind> functionsToReduce;
+
+  /**
+   * Gets an instance of AggregateReduceFunctionsRule
+   * with client provided specific functions to reduce
+   * @param functionsToReduce client provided information
+   *                          on which specific functions will be
+   *                          reduced by this rule
+   * @return an instance of AggregateReduceFunctionsRule
+   */
+  public static AggregateReduceFunctionsRule
+      getInstanceWithSpecificFunctionsToReduce(final Set<SqlKind> functionsToReduce) {
+    Preconditions.checkArgument(functionsToReduce != null,
+        "Error: expecting a valid handle for AggregateFunctionsToReduce");
+    return new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
+      RelFactories.LOGICAL_BUILDER, functionsToReduce);
+  }
+
   //~ Constructors -----------------------------------------------------------
 
   /** Creates an AggregateReduceFunctionsRule. */
   public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
       RelBuilderFactory relBuilderFactory) {
     super(operand, relBuilderFactory, null);
+    // by default, this rule will reduce all functions it handles
+    this.functionsToReduce = new HashSet<>();
+    addDefaultSetOfFunctionsToReduce();
+  }
+
+  /**
+   * Creates an AggregateReduceFunctionsRule with client
+   * provided information on which specific functions will
+   * be reduced by this rule
+   * @param operand root operand
+   * @param relBuilderFactory builder for relational expressions
+   * @param functionsToReduce client provided information
+   *                          on which specific functions
+   *                          will be reduced by this rule
+   */
+  public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
+      RelBuilderFactory relBuilderFactory, Set<SqlKind> functionsToReduce) {
+    super(operand, relBuilderFactory, null);
+    this.functionsToReduce = functionsToReduce;
   }
 
   //~ Methods ----------------------------------------------------------------
 
+  private void addDefaultSetOfFunctionsToReduce() {
```

Comment:
Minor but why not reusing org.apache.calcite.sql.SqlKind#AVG_AGG_FUNCTIONS and org.apache.calcite.sql.SqlKind#COVAR_AVG_AGG_FUNCTIONS since they exist?

Reply:
done.